### PR TITLE
namespace clients

### PR DESCRIPTION
**Pull request by anwar6953**
I needed this change to make it easier to demo namespaces of streams.
Dataset-related clients are not being namespaced in this PR, as the entities are not namespaced in CDAP.

Build: http://builds.cask.co/browse/CDAP-DUT801-5

### CODE REVIEW

**Comment by jwang47** [comment]
LGTM

**Comment by anwar6953** [comment]
Moved namespace commands into its own category.
`use namespace X` will now fail if namespace X doesn't exist.

File: cdap-cli/src/main/java/co/cask/cdap/cli/command/UseNamespaceCommand.java [hunk 1]
```diff
@@ -31,16 +32,21 @@
...
     super(cliConfig);
     this.cliConfig = cliConfig;
+    this.namespaceClient = namespaceClient;
   }
 
   @Override
   public void perform(Arguments arguments, PrintStream output) throws Exception {
     String namespace = arguments.get(ArgumentName.NAMESPACE_ID.toString());
+    // Check if namespace exists; throws exception if namespace doesn't exist.
+    namespaceClient.get(namespace);
```
Comment (nitinmotgi) [inline cdap-cli/src/main/java/co/cask/cdap/cli/command/UseNamespaceCommand.java]: CLI should not show exception to user. Will it show proper error message saying "Failed to change to <namespace> as the <namepsace> doesn't exist"

**Comment by jwang47** [comment]
LGTM

File: cdap-cli/src/main/java/co/cask/cdap/cli/command/UseNamespaceCommand.java (hunk 1 above)
Comment (anwar6953) [inline cdap-cli/src/main/java/co/cask/cdap/cli/command/UseNamespaceCommand.java]: Yea, it presents the error properly:

```
cdap (http://127.0.0.1:10000)> use namespace bar

use namespace bar
Error: namespace 'bar' was not found
```

**Comment by nitinmotgi** [comment]
LGTM :+1:

**Pull request merged**